fuel-p2p/src/discovery/discovery_config.rs

```diff
@@ -117,16 +137,31 @@ impl DiscoveryConfig {
             })
             .collect::<Vec<_>>();
 
-        for (peer_id, address) in &bootstrap_nodes {
+        // reserved nodes need to have their peer_id defined in the Multiaddr
+        let reserved_nodes = reserved_nodes
+            .into_iter()
+            .filter_map(|node| {
+                PeerId::try_from_multiaddr(&node).map(|peer_id| (peer_id, node))
+            })
+            .collect::<Vec<_>>();
+
+        // add bootstrap nodes only if `reserved_nodes_only_mode` is disabled
+        if !reserved_nodes_only_mode {
+            for (peer_id, address) in &bootstrap_nodes {
+                kademlia.add_address(peer_id, address.clone());
+            }
+        }
+
+        for (peer_id, address) in &reserved_nodes {
             kademlia.add_address(peer_id, address.clone());
         }
 
         if let Err(e) = kademlia.bootstrap() {
             warn!("Kademlia bootstrap failed: {}", e);
         }
 
-        let next_kad_random_walk = if self.with_random_walk {
-            Some(Delay::new(Duration::new(0, 0)))
+        let next_kad_random_walk = if self.with_random_walk && !reserved_nodes_only_mode {
+            Some(Delay::new(Duration::from_secs(5)))
```

Comment:
Might be worth making this a const and having a comment on why it's 5 seconds?

Reply:
good point, this was unrelated to the issue itself, I just noticed that random walk was set to 0 will update it with a configurable value